aioxmpp/avatar/service.py

```diff
@@ -503,41 +712,12 @@ def subscribe(self, jid):
         """
         yield from self._pubsub.subscribe(jid, namespaces.xep0084_metadata)
 
-    @asyncio.coroutine
-    def _check_for_pep(self):
-        # determine support for PEP as specified in XEP-0163 section 6
-        # XXX: fix this by implementing a PEPService that is derived from
-        # pubsub and checks for the server capability and simplifies the
-        # handling
-        def raise_exception():
-            raise NotImplementedError(
-                "Server does not support PEP and we do not support "
-                "surrogating for lack of PEP support"
-            )
-
-        if self._has_pep is not None:
-            if self._has_pep:
-                return
-            else:
-                raise_exception()
-
-        disco_info = yield from self._disco.query_info(
-            self.client.local_jid.bare()
-        )
-
-        for item in disco_info.identities.filter(attrs={"category": "pubsub"}):
-            if item.type_ == "pep":
-                self._has_pep = True
-                break
-        else:
-            self._has_pep = False
-            raise_exception()
-
     @aioxmpp.service.depsignal(aioxmpp.stream.StanzaStream,
                                "on_stream_destroyed")
     def handle_stream_destroyed(self, reason):
-        # invalidate the cache
-        self._has_pep = None
+        # invalidate the cache?
+        self._vcard_ressource_interference.clear()
+        self._has_pep_avatar.clear()
```

Comment:
Metadata cache?

Reply:
I believe we discussed this issue before and agreed not to wipe the cache, but I am still not sure which behaviour is correct.